torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -108,7 +108,7 @@ def _broadcast_object(
         )
         dist.broadcast(data_recv_tensor, src=src_rank, group=group, async_op=False)
         buffer = io.BytesIO(data_recv_tensor.cpu().numpy())
-        obj = torch.load(buffer, map_location=device)
+        obj = torch.load(buffer, map_location=device, weights_only=False)
```

Comment:
From the code above, we can see this is broadcasting an object pickled to bytes, so loading here means unpickling. The function doesn't make any assumptions like obj being a tensor, so we need to set `weights_only=False` here.